### PR TITLE
Target Node.js 18, update dependencies

### PR DESCRIPTION
**Pull request by tommy-mitchell**
This PR:

- Targets Node.js 18
- Updates dependencies (specifically to get [`strip-ansi` v7.1.0](https://github.com/chalk/strip-ansi/releases/tag/v7.1.0))
- Improves tests, covering a previously untested case

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
 });
+
+test('no input', async t => {
+	/** @type {import('execa').ExecaError} */
+	const error = await t.throwsAsync(execa('./cli.js', {stdin: 'inherit'}));
+
+	t.like(error, {
+		stderr: 'Input required',
+		exitCode: 1,
+	});
```
Comment (tommy-mitchell) [inline test.js:24]: ...yeah, I guess it makes sense that the action runner's shell isn't TTY lol

Does execa have a way to simulate it?

Comment (sindresorhus) [reply test.js:24]: No, but you can force Chalk to use colors even when not interactive using an environment variable.

Comment (tommy-mitchell) [reply test.js:24]: https://github.com/chalk/strip-ansi-cli/blob/5ed811e7dac35b2b1759eef570f73c1e63b82d64/cli.js#L24-L27

It’s for this check, I’m not sure if that would help.

Comment (sindresorhus) [reply test.js:24]: Ah, ok. You could use a PTY, but not worth the effort. Just skip that test.

Comment (tommy-mitchell) [reply test.js:24]: So this workaround worked: https://github.com/tommy-mitchell/strip-ansi-cli/commit/a8bb1594b9ed312af19f053d6e294819d88bd7d7

From https://github.com/actions/runner/issues/241#issuecomment-577360161

**Pull request merged**